src/services/github.ts

```diff
@@ -142,6 +143,35 @@ export async function githubCreate(context: ContextType<typeof AppContext>, base
     return data;
 }
 
+export async function githubUpdate(context: ContextType<typeof AppContext>, basePath: string, name: string, initialContent: string, sha: string, repo: GitHubRepository = "content") {
+    const path = `${basePath}/${name}`;
+
+    // If we have a binary file, we want to do the conversion as the binary file, so use the standard btoa
+    // But if there are any >255 characters in there, this must be UTF text so we use the encoder that
+    // first turns UTF-16 into UTF-8 as UTF-16 can't be encoded as base64 (since some "bytes" are > 255).
+    let content;
+    try {
+        content = window.btoa(initialContent);
+    } catch (e) {
+        content = encodeBase64(initialContent);
+    }
+
+    const data = await fetcher(contentsPath(path, undefined, repo), {
+        method: "PUT",
+        body: {
+            branch: context.github.branch,
+            message: "Updating " + path,
```

Comment:
Because of the way we're using this I'd be tempted to say "Replacing " or "Replaced " and edit the function's name to `githubReplace(...)` too. Otherwise it is hard to differentiate between "Updating " and "Edited ".